pkg/pusher/pusher.go

```diff
@@ -226,6 +228,9 @@ func (s *Service) chunksWorker(warmupTime time.Duration, tracer *tracing.Tracer)
 	for {
 		select {
 		case op := <-cc:
+			if s.inflight.set(op.Chunk) {
```

Comment:
would a chunk ever be attempted to be pushed twice?
do we need this inflight anymore?

Reply:
So I had initially removed this, but there is a case for it. I have removed the complexity in the Iteration by which it will restart from start every time its gets a trigger. The item will only be removed from the pushIndex if its reported to be synced/stored. As a result it can be sent to pusher again while an attempt is in progress. This is the easiest way to handle this case.